encodings/bytebool/src/array.rs

```diff
@@ -118,7 +118,7 @@ impl From<Vec<bool>> for ByteBoolArray {
 
 impl From<Vec<Option<bool>>> for ByteBoolArray {
     fn from(value: Vec<Option<bool>>) -> Self {
-        let validity = Validity::from_iter(value.iter());
+        let validity = Validity::from_iter(value.iter().cloned());
```

Comment:
why `cloned()`? it just needs to call `is_some()` which can be done by ref

Reply:
oh, this could bu just `value.into_iter()`

Reply:
Yeah, good catch. I might remove this FromIterator since it's easy to misuse vs FromIterator<bool>, or at least one of the Froms...

Reply:
The value iterator is used twice. This was actually using the confusing `FromIterator<Option<E>> for Validity`, I think it was the only place... and it's more confusing that E is boolean. So now it uses FromIterator<bool> explicitly and calls is_some() on the values.